src/rawArray/ops.ts

```diff
@@ -0,0 +1,114 @@
+import { ArraySelection, SliceIndices } from '../core/types';
+import { normalizeArraySelection, selectionToSliceIndices } from '../core/indexing';
+import { TypedArray } from '../nestedArray/types';
+
+export function setRawArrayToScalar(dstArr: TypedArray, dstStrides: number[], dstShape: number[], dstSelection: number | ArraySelection, value: number) {
+    // This translates "...", ":", null, etc into a list of slices.
+    const normalizedSelection = normalizeArraySelection(dstSelection, dstShape, true);
+    const [sliceIndices] = selectionToSliceIndices(normalizedSelection, dstShape);
+    // Above we force the results to be SliceIndicesIndices only, without integer selections making this cast is safe.
+    _setRawArrayToScalar(value, dstArr, dstStrides, sliceIndices as SliceIndices[]);
+}
+
+export function setRawArrayDirect(dstArr: TypedArray, dstStrides: number[], dstShape: number[], dstSelection: number | ArraySelection, sourceArr: TypedArray, sourceStrides: number[], sourceShape: number[], sourceSelection: number | ArraySelection) {
+    // This translates "...", ":", null, etc into a list of slices.
+    const normalizedDstSelection = normalizeArraySelection(dstSelection, dstShape, true);
+    // Above we force the results to be dstSliceIndices only, without integer selections making this cast is safe.
+    const [dstSliceIndices] = selectionToSliceIndices(normalizedDstSelection, dstShape);
+
+    const normalizedSourceSelection = normalizeArraySelection(sourceSelection, sourceShape, false);
+    const [sourceSliceIndicies] = selectionToSliceIndices(normalizedSourceSelection, sourceShape);
+
+    _setRawArrayDirect(dstArr, dstStrides, dstSliceIndices as SliceIndices[], sourceArr, sourceStrides, sourceSliceIndicies);
+}
+
+function _setRawArrayDirect(dstArr: TypedArray, dstStrides: number[], dstSliceIndices: SliceIndices[], sourceArr: TypedArray, sourceStrides: number[], sourceSliceIndicies: (SliceIndices | number)[]) {
+    if (sourceSliceIndicies.length === 0) {
+        // Case when last source dimension is squeezed
+        dstArr.set(sourceArr);
+        return;
+    }
+
+    // Get current indicies and strides for both destination and source arrays
+    const [currentDstSlice, ...nextDstSliceIndicies] = dstSliceIndices;
```

Comment:
Nice, I think I just used `x = list[0]` and `xNext = list.slice(1)`, but this is much nicer

Reply:
Yeah, it took me some time to figure out what was going on, so I used destructuring and long variable names to help me remember what was going on 😅